src/mpid/ch4/generic/mpidig_init.c

```diff
@@ -18,6 +18,7 @@ int MPIDIG_init(void)
 
     MPIDI_global.is_ch4u_initialized = 0;
     MPIDIG_am_init();
+    MPIDIG_gpu_init();
     MPIDIU_map_create((void **) &(MPIDI_global.win_map), MPL_MEM_RMA);
     MPIDI_global.csel_root = NULL;
```

Comment:
I am surprised this is here. In fact, probably the entire `MPIDIG_init` should just be in `ch4_init`. There is no relationship between `MPIDIG_am` and `MPIDIG_gpu`. They are just two "generic" components that can be independently used.

Reply:
You are right the two are independent but still why should not these be initialized in a generic init?